src/main/java/de/ustutt/iaas/cc/core/RemoteTextProcessorMultiLeastConnection.java

```diff
@@ -0,0 +1,79 @@
+package de.ustutt.iaas.cc.core;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.PriorityQueue;
+import java.util.concurrent.CopyOnWriteArrayList;
+import java.util.concurrent.CopyOnWriteArraySet;
+
+import javax.ws.rs.client.Client;
+import javax.ws.rs.client.Entity;
+import javax.ws.rs.client.WebTarget;
+import javax.ws.rs.core.MediaType;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class RemoteTextProcessorMultiLeastConnection implements ITextProcessor {
+
+	private Client client;
+	// TODO Make queue Concurrent!
+	private CopyOnWriteArrayList<Endpoint> endpointList;
+	
+	private final static Logger logger = LoggerFactory.getLogger(RemoteTextProcessorMultiLeastConnection.class);
+
+	public RemoteTextProcessorMultiLeastConnection(List<String> eps, Client clientm) {
+		super();
+		this.client = clientm;
+		this.endpointList = new CopyOnWriteArrayList<Endpoint>();
+		eps.forEach(ep -> endpointList.add(new Endpoint(ep)));
+	}
+
+	@Override
+	public String process(String text) { 
+		synchronized (this) {
+			logger.debug(endpointList.toString());
+			Collections.sort(endpointList);
```

Comment:
According to the JavaDoc of `CopyOnWriteArrayList`:

> This is ordinarily too costly, but may be more efficient than alternatives when traversal operations vastly outnumber mutations,...

Your code does only read the list, but I would assume that the implementation of `Collections.sort(...)` will modify the list a lot, i.e. this might result in a serious performance issue here.